test/Test/Pos/Util/ModifierSpec.hs

```diff
@@ -0,0 +1,185 @@
+{-# LANGUAGE GADTs #-}
+
+-- | Specification for Pos.Util.Modifier (from 'core')
+
+module Test.Pos.Util.ModifierSpec
+       ( spec
+       ) where
+
+import qualified Data.HashMap.Strict   as HM
+
+import qualified Pos.Util.Modifier     as Core
+import           Universum
+
+import           Test.Hspec            (Spec, describe)
+import           Test.Hspec.QuickCheck (prop)
+import           Test.Pos.Util         (formsMonoid)
+import           Test.QuickCheck       (Property, (==>))
+
+spec :: Spec
+spec = describe "MapModifier" $ do
+    describe "Monoid instance" $ do
+        prop
+            "Map modifiers form a monoid instance with the custom binary operation"
+            (mapModifierFormsMonoid @Text @Text)
+    describe "Operations on map" $ do
+        describe "insert" $ do
+            prop
+                "inserting a value and then looking it up returns the inserted value"
+                (insertThenLookup @Text @Text)
+        describe "delete" $ do
+            prop
+                "inserting a value and then deleting it removes it from the map"
+                (insertDeleteThenLookup @Text @Text)
```

Comment:
Btw, there are two different cases:
• something is inserted into underlying map and deleted from modifier;
• something is inserted into modifier and then deleted from it.

I think it would be good to cover both cases and clarify description.